zip-bolt-support.rst

```diff
@@ -0,0 +1,408 @@
+
+3. Custom Shielded Tx: Using Z-addresses and Scriptless
+-------------
+We assume the following features are present:
+
+(a) ``lock_time`` - for absolute lock time
+(b) A way to enforce relative lock time
+(c) 2-of-2 multi-sig shielded address support
+(d) All inputs/outputs are specified from/to a shielded address
+(e) A method to encumber the outputs of a shielded transaction
+(f) An extension to the transaction format to include BOLT (e.g., like ``vBoltDescription``)
+(g) Extend the ``SIGHASH`` flags to cover the extended field
+
+The goal here is to perform all the same validation steps for channel opening/closing without relying on the scripting system, as well as allowing for relative timelocks (the equivalent of ``OP_CSV``). In order to support multihop payments, we need absolute timelocks as well (the equivalent of ``OP_CLTV``). We also want to ensure that transactions are non-malleable in order to allow for unconfirmed dependency transaction chains.
+
+**Limitations/Notes**: With extensions to shielded transaction format, it may be evident whenever parties are establishing private payment channels. We appreciate feedback on the feasibility of what is proposed for each aspect of the Bolt protocol.
+
+3.1 Channel Opening
+-------------
+The customer creates a funding transaction that spends ZEC from a shielded address to a 2-of-2 multi-sig shielded address. Here is the flow (1) creating a multisig shielded address specifying both parties keys and (2) generating channel tokens.
+
+3.2 Funding Transaction
+-------------
+The funding transaction is by default funded by only one participant, the customer. It could also be funded by the merchant. 
+
+This transaction has 2 shielded inputs (but can be up to some N) and 1 output to a 2-of-2 shielded address with the merchant public key. If an ``vBoltDescription`` field is added, then we could use it to store the channel parameters and the channel token for opening the channel.
+
+3.3 Initial Wallet Commitment
+-------------
+The initial wallet commitment will spend from the shielded address to two outputs: a P2SH output (for customer) and P2PKH (for merchant).  The first output pays the customer with a timelock (or the merchant with a revocation token) and the second output allows the merchant to spend immediately. It is not clear to us whether it will be possible to encumber the outputs of shielded outputs directly. 
+
+We would appreciate feedback on the possibilities with creating commitment transactions via shielded transactions only.
+
+3.4. Channel Closing
+-------------
+The channel closing consists of the customer broadcasting the most recent commitment transaction and requires that they present the closure token necessary to claim the funds. Similarly, the merchant would be able to claim the funds with the appropriate revocation token as well.
+
+4. Bitcoin Compatible: Using T-address and Scripting Opcodes
```

Comment:
Remove this section from the ZIP. Zcash does not have `P2WSH` addresses (and it is unlikely that they would ever be added), and IMHO a transparent-only BOLT integration is not worth the non-trivial engineering effort, when that time could instead be spent working on one of the other proposals with better privacy properties.

Reply:
Note: don't actually remove this section until another reviewer agrees with me that it should be removed!